tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -312,6 +325,42 @@ private int WriteSpanLink(ref byte[] bytes, int offset, in SpanModel spanModel)
             return offset - originalOffset;
         }
 
+        private int WriteSpanEvent(ref byte[] bytes, int offset, in SpanModel spanModel)
+        {
+            int originalOffset = offset;
+
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _spanEventBytes);
+            offset += MessagePackBinary.WriteArrayHeader(ref bytes, offset, spanModel.Span.SpanEvents.Count);
+
+            foreach (var spanEvent in spanModel.Span.SpanEvents)
+            {
+                var len = 3; // name, timestamp, and attributes
+                offset += MessagePackBinary.WriteMapHeader(ref bytes, offset, len);
+
+                // Write name
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _nameBytes);
+                offset += MessagePackBinary.WriteString(ref bytes, offset, spanEvent.Name);
+
+                // Write timestamp
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _timeUnixNanoBytes);
+                offset += MessagePackBinary.WriteInt64(ref bytes, offset, TimeExtensions.ToUnixTimeNanoseconds(spanEvent.Timestamp ?? DateTimeOffset.UtcNow));
+
+                // Write attributes if any
+                if (spanEvent.Attributes is { Count: > 0 })
+                {
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _attributesBytes);
+                    offset += MessagePackBinary.WriteMapHeader(ref bytes, offset, spanEvent.Attributes.Count);
+                    foreach (var attribute in spanEvent.Attributes)
+                    {
+                        offset += MessagePackBinary.WriteString(ref bytes, offset, attribute.Key);
+                        offset += MessagePackBinary.WriteString(ref bytes, offset, attribute.Value);
```

Comment:
Our RFC says we have to represent the attribute value in a certain way. You'll want to double check with @marcotc but I believe to generate an attribute with a string value you'll need to replace the string with the equivalent object:

```
{
    "type": 0,
    "string_value": "<value>"
}
```

So if that's correct, then you're probably going to have to do the following:
- `MessagePackBinary.WriteMapHeader` with a size of two
- `MessagePackBinary.WriteString` for the `type` string
- `MessagePackBinary.WriteInt64` with a value of `0`
- `MessagePackBinary.WriteString` for the `string_value` string
- `MessagePackBinary.WriteString` for the actual `attribute.Value`